### PR TITLE
Linear onboarding milestone 2 - Part 2: Clean up unused code in AppEntryActivity

### PR DESCRIPTION
**Pull request by gizemb**
### Issues

This PR cleans up some dead code inside AppEntryActivity. I've also checked with 3.44 version, and seems like these methods are not used for some time.

It also adds a static factory method for AppEntryActivity, to ensure consistency when starting this activity.

### Testing

Manually tested the login functionality best to my knowledge, but QA Automation tests should reveal if anything is missed.

#### APK
[Download build #1120](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1120/artifact/build/artifact/wire-dev-PR2581-1120.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/appentry/fragments/TeamNameFragment.scala [hunk 1]
```diff
@@ -56,5 +56,7 @@ case class TeamNameFragment() extends CreateTeamFragment with SSOFragment {
 }
 
 object TeamNameFragment {
-  val Tag: String = getClass.getSimpleName
+  val Tag: String = "TeamNameFragment"
+
```
Comment (mejdoo) [inline app/src/main/scala/com/waz/zclient/appentry/fragments/TeamNameFragment.scala]: you can remove the empty line here

Comment (gizemb) [reply app/src/main/scala/com/waz/zclient/appentry/fragments/TeamNameFragment.scala]: why? Wouldn't you normally separate a field declaration and a method declaration, in a regular class?

Comment (mejdoo) [reply app/src/main/scala/com/waz/zclient/appentry/fragments/TeamNameFragment.scala]: alright

**Comment by gizemb** [comment]
build again

**Comment by gizemb** [comment]
build again

**Comment by gizemb** [comment]
build again

**Comment by mejdoo** [comment]
build again

**Pull request merged**